system/src/system.cpp

```diff
@@ -66,6 +69,52 @@ int lua_threadcount(lua_State* L)
     return 1;
 }
 
+int lua_freememory(lua_State* L)
+{
+    lua_pushinteger(L, uv_get_free_memory()); // FIXME: this is a 64 bit int, deal with it?
```

Comment:
Use `lua_pushnumber` here, casting it to a double. Divide the uint64_t by the size of a megabyte, so it's in megabytes.

This will have enough precision.